java/dataset/src/main/java/org/apache/arrow/dataset/scanner/ScanOptions.java

```diff
@@ -58,6 +60,18 @@ public ScanOptions(long batchSize, Optional<String[]> columns) {
     this.columns = columns;
   }
 
+  /**
+   * Constructor.
+   * @param batchSize Maximum row number of each returned {@link org.apache.arrow.vector.ipc.message.ArrowRecordBatch}
+   * @param substraitExtendedExpression Extended expression to evaluate for project new columns or apply filter.
+   */
+  public ScanOptions(long batchSize, ByteBuffer substraitExtendedExpression) {
```

Comment:
We can and still should have a builder, I was just saying that it doesn't make sense to pass Optional as a constructor parameter here...

Reply:
Oh, ok, let me consider the filter again

Reply:
added